File: ios/NHSOnline/NHSOnline/Data/Configuration.swift
import Foundation

struct Configuration: Codable {
    let fidoServerUrl: String
    let minimumSupportediOSVersion: String
    let knownServices: [RootService]
    let nhsLoginLoggedInPaths: [String]

    init(fidoServerUrl: String, minimumSupportediOSVersion: String, knownServices: [RootService], nhsLoginLoggedInPaths: [String]) {
        self.fidoServerUrl = fidoServerUrl
        self.minimumSupportediOSVersion = minimumSupportediOSVersion
        self.knownServices = knownServices
        self.nhsLoginLoggedInPaths = nhsLoginLoggedInPaths
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/services/Configuration.kt
package com.nhs.online.nhsonline.services

import com.nhs.online.nhsonline.services.knownservices.RootService
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Configuration (
    var minimumSupportedAndroidVersion: String,
    var fidoServerUrl: String,
    var knownServices: List<RootService>,
    var nhsLoginLoggedInPaths: List<String>?
)
